src/server/controllers/find-address.controller.js

```diff
@@ -52,12 +52,13 @@ const findAddressController = async (currentPage, previousAnswers, newAnswers) =
 
     controllerResponse.addressLookups[searchPostcodeFieldName] = addressesForPostcode;
 
+    //the below used to take the current page and add the suffix '-select- or '-manual', this is because the establishment address type page was incorrectly appearing as establishment-address and thus skipping the page validation, the below removes the -type and keeps the path as expected
     if (addressesForPostcode.length > 0) {
-      controllerResponse.switches[`${currentPage}-none-found`] = false;
-      controllerResponse.redirectRoute = `${currentPage}-select`;
+      controllerResponse.switches[`${currentPage.replace("-type", "")}-none-found`] = false;
```

Comment:
Rather than repeating this 4 times, personally I would find it clearer to use a new variable with a simple comment where it is set:
```
let addressType = currentPage;
if (currentPage == "establishment-address-type") {
    // find-address for establishments is triggered by the establishment-address-type page 
    addressType = "establishment-address";
}
```